crates/iceberg/src/catalog/mod.rs

```diff
@@ -133,6 +133,16 @@ impl NamespaceIdent {
     pub fn inner(self) -> Vec<String> {
         self.0
     }
+
+    /// Get the parent of this namespace.
+    /// Returns None if this namespace only has a single element and thus has no parent.
+    pub fn parent(&self) -> Option<Self> {
+        if self.0.len() <= 1 {
+            None
+        } else {
+            Some(Self(self.0[..self.0.len() - 1].to_vec()))
+        }
+    }
```

Comment:
How about using [split_last](https://doc.rust-lang.org/stable/std/primitive.slice.html#method.split_last)? So that we have:

```rust
pub fn parent(&self) -> Option<Self> {
    let (parent, _) = self.0.split_last()?;
    Some(Self(parent.to_vec()))
}
```

Reply:
It's not quite as elegant, as split_last() might still yield an empty parent. I switched to `split_last` in the latest commit.

Reply:
You are right. I misremembered this API.